### ISSUE TITLE
`approval-voting`: experiment with batching signature checks

### ISSUE THREAD

**Issue by sandreim** [paritytech/polkadot#6613]
We could additionally gain some performance by batching signature checks. Right now we check them individually by handling `CheckAndImportAssignment` and `CheckAndImportApproval`. We have to test if there are any significant gains by doing something like, buffering signature checks until reach a limit or N milliseconds have passed since we started buffering.  This would play nicely with https://github.com/paritytech/polkadot/issues/6610.

Another approach is to do the batching in `approval-distribution` but I am not really sure if this would be more effective.

**Comment by burdges** [comment]
I vaguely recalled "batching" here does not mean "cryptographic batching" like I mentioned in #6613?

**Comment by sandreim** [comment]
Yep, it is actually not, but stacks with "cryptographic batching" as we scale up validator and parachain count. It's mostly about improving our architecture of processing approval voting gossip messages and slicing the CPU intensive work.

**Comment by burdges** [comment]
cc #7554
